Create a documentation style guide to apply across docs (#325)
mckinsey/vizro: vizro-core/docs/pages/development/documentation-style-guide.md

# Vizro documentation style guide

This is the style guide we apply to the [Vizro documentation](https://vizro.readthedocs.io/en/stable/).

We ask anyone kind enough to contribute documentation changes to follow this style for consistency and simplicity.

What follows is a set of lightweight guidelines rather than rules. There are always edge cases and exceptions, and if it's not obvious what the style should be, consult the [Microsoft style guide](https://docs.microsoft.com/en-gb/style-guide/welcome/) for an example of good practice. We also use the [INCITS Inclusive Terminology Guidelines](https://standards.incits.org/apps/group_public/download.php/131246/eb-2021-00288-001-INCITS-Inclusive-Terminology-Guidelines.pdf).

## Vizro lexicon

The names of our products are **Vizro** and **Vizro-AI**.

> We refer to other products using their preferred capitalization. For example, Dash and Plotly are always capitalized, except where given as Python package names `dash` and `plotly`.

Reviewer: ```suggestion
We refer to other products using their preferred capitalization. For example, Dash and Pydantic are always capitalized, except where given as Python package names `dash` and `pydantic`.
```

I more often find myself altering capitalisation of `pydantic` than `plotly` in our docs.

Vizro components are named using lower case:

> > "Here is a guide to using containers..."

Reviewer: ```suggestion
> Here is a guide to using containers...
```

(comment applies in a few places)

Unless you meant these to be here for some particular reason? They seem to be in some quotes but not others.

Use code font when referring to the component as a class or object:

> "To add a `Container` to your page..."

> 

Reviewer: Fussy point but something I find myself editing quite a bit. What happens if you pluralise these? Is it `Containers` or `Container`s? I go for the latter but we're not consistent, and maybe it's too fiddly to enforce.

Author: It's annoying isn't it -- I don't like to be pernickety about it either. If possible, without overt textual gymnastics, I'd write around it. So use the phrase `Container` objects rather than `Container`s or something, but only if it make sense to do it and doesn't make the text more cluttered.

## Bullets
* Capitalize the first word, and end the bullet with a period.
* Don't use numbered bullets except for a sequence of instructions, or where you have to refer back to one of them in the text (or a diagram).

## Call out boxes

Keep the amount of text, and the number and variety of callouts used, to a minimum. There is a [broad set available](https://squidfunk.github.io/mkdocs-material/reference/admonitions/#supported-types) for use in the Vizro docs, but we prefer to limit usage mostly to the following:

!!!note "note"

>      For notable information

Reviewer: ```suggestion
     For notable information.
```

!!!warning "warning"

     To indicate a potential gotcha.

!!!example "example"

>      For example code

Reviewer: ```suggestion
     For example code.
```

> Boxes can be made collapsible, and rendered initially collapsed or expanded.

Reviewer 2: ```suggestion
Boxes can be made collapsible, and if done so preferably should be rendered initially collapsed.
```
Point for discussion - but I think it's a good light rule. Make things collapsible if they are crowding the page, and hide them initially, do not make things collapsible that should always be visible

> Boxes can be made collapsible, and if you choose this option, prefer to add them to the page so they are initially collapsed, to avoid crowding.

Reviewer: I would have thought that this is pretty dependent on the sort of callout, e.g. don't we want warnings to be expanded by default?

Might also be worth saying how in Markdown you make a box collapsible or at least linking to mkdocs explanation of it.

## Capitalization
* Only capitalize proper nouns e.g. names of technology products, other tools and services.
* Don't capitalize cloud, internet, machine learning, or advanced analytics. Take a look at the [Microsoft style guide](https://docs.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/term-collections/accessibility-terms) if you're unsure.

> * Follow sentence case, which capitalizes only the first word of a title/subtitle. We prefer this "An introduction to data visualization" to this "An Introduction to Data Visualization".

Reviewer: ```suggestion
* Follow sentence case, which capitalizes only the first word of a title/subtitle. We prefer "An introduction to data visualization" to "An Introduction to Data Visualization".
```

## Code formatting
* Mark code blocks with the appropriate language to enable syntax highlighting.
* We use a `bash` lexer for all codeblocks that represent the terminal, and we don't include the prompt.

> * Use the code format for Python package names such as `pandas` or `pylint`.

Reviewer: ```suggestion
* Use the code format for Python package names such as `pandas` or `pydantic`.
```

## Instructions

Prefer to use imperatives to make instructions. For example:
> "Complete the configuration steps"

You don't need to use the word "please" -- readers want less to read and don't think it's rude if you omit it.

You can also use second person:

> > "You should complete the configuration steps". Don't use the passive "The configuration steps should be completed" (see next bullet).

Reviewer: This looks strangely formatted  - should the "don't use the passive" be in here? Also not sure what "see next bullet" refers to?

Avoid using the passive tense where possible.

!!!note "What is passive tense?"

    If you can add "by zombies" to the end of any sentence, it is passive.

    * For example: "The configuration steps should be completed." can also be read as: "The configuration should be completed BY ZOMBIES".

>     * Instead, you'd write this: "You should complete the configuration steps" or better still, "Complete the configuration steps".

Reviewer 3: This is so funny 🤣  I love this!! 🚀

## Language
* Use US English.

## Links
* Make hyperlink descriptions as descriptive as you can. This is a good description:

> Learn how to [contribute to Vizro](https://vizro.readthedocs.io/en/stable/pages/development/contributing/).

This is less helpful:

> [Learn how to](https://vizro.readthedocs.io/en/stable/pages/development/contributing/) contribute to Vizro.

Don't write this:

> To learn how to contribute to Vizro, see [here](https://vizro.readthedocs.io/en/stable/pages/development/contributing/).

## Oxford commas
Use these in lists to avoid confusion. This is confusing:

> The ice cream comes in a range of flavors including banana and strawberry, mango and raspberry and blueberry.

This is clearer:

> The ice cream comes in a range of flavors including banana and strawberry, mango and raspberry, and blueberry.

## Style

Keep your sentences short and easy to read. Your tone of voice should be simple, friendly and functional.

### Simple

Simple is clear, concise and direct:

> > We build new skills within your team.

Reviewer: ```suggestion
> We build new skills in your team.
```

even simpler?

Simple is not vague, verbose or full of jargon:

> > We leverage your existing organizational resources to synthesize novel competencies.

Reviewer: I feel like this guidance conflicts heavily with McKinsey tradition 😆
